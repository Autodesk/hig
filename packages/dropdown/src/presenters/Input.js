import React from "react";
import Icon from "@hig/icon";
import { TextFieldPresenter } from "@hig/text-field";
import "@hig/icon/build/index.css";
import "@hig/text-field/build/index.css";

import "./Input.scss";

export default function Input(props) {
  return (
    <div className="hig__dropdown__input-wrapper">
      <TextFieldPresenter {...props} type="text" readOnly />
      <span className="hig__dropdown__input-caret">
        {/* @TODO: there are variations of the TextField with multiple icons at the end of the input. These icon nodes should be passed as props to TextField. */}
        <Icon name="caret" />
      </span>
    </div>
  );
}

Input.propTypes = {
  ...TextFieldPresenter.propTypes,
  type: undefined,
  readOnly: undefined
};
