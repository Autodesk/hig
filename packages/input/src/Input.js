import React, { Component } from "react";
import PropTypes from "prop-types";

import InputBehavior from "./InputBehavior";
import InputPresenter from "./InputPresenter";

class Input extends Component {
  static propTypes = {
    value: PropTypes.string,
    disabled: PropTypes.bool,
    onChange: PropTypes.func
  };

  render() {
    const { value, disabled, onChange } = this.props;

    return (
      <InputBehavior>
        {({
          hasHover,
          onMouseEnter,
          onMouseLeave,
          hasFocus,
          onFocus,
          onBlur
        }) => (
          <InputPresenter
            value={value}
            disabled={disabled}
            onFocus={onFocus}
            onBlur={onBlur}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            onChange={onChange}
            hasHover={hasHover}
            hasFocus={hasFocus}
          />
        )}
      </InputBehavior>
    );
  }
}

export default Input;
