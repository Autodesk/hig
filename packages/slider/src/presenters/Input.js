import React, { Component } from "react";
import PropTypes from "prop-types";

import "./input.scss";

export default class Input extends Component {
  static propTypes = {
    /**
     * Prevents user actions on the field
     */
    disabled: PropTypes.bool,
    /**
     * HTML ID attribute
     */
    id: PropTypes.string,
    /**
     * Minimum value of the slider
     */
    min: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /**
     * Maximum value of the slider
     */
    max: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /**
     * Name of the field when submitted with a form
     */
    name: PropTypes.string,
    /**
     * Called when user moves focus from the field
     */
    onBlur: PropTypes.func,
    /**
     * Called after user changes the value of the field
     */
    onChange: PropTypes.func,
    /**
     * Called when user puts focus onto the field
     */
    onFocus: PropTypes.func,
    /**
     * Called as user changes the value of the field
     */
    onInput: PropTypes.func,
    /**
     * The granularity of each step on the slider
     */
    step: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /**
     * Value of the field
     */
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  };

  render() {
    const { min, max, value } = this.props;

    return (
      <input
        className="hig__slider__input"
        type="range"
        aria-valuemin={min}
        aria-valuemax={max}
        aria-valuenow={value}
        {...this.props}
      />
    );
  }
}
