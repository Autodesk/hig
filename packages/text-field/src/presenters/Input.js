import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";

import "./input.scss";

export default class Input extends Component {
  static propTypes = {
    /**
     * Prevents user actions on the field
     */
    disabled: PropTypes.bool,
    /**
     * HTML ID attribute
     */
    id: PropTypes.string,
    /**
     * Name of the field when submitted with a form
     */
    name: PropTypes.string,
    /**
     * Called when user moves focus from the field
     */
    onBlur: PropTypes.func,
    /**
     * Called after user changes the value of the field
     */
    onChange: PropTypes.func,
    /**
     * Called when user puts focus onto the field
     */
    onFocus: PropTypes.func,
    /**
     * Called as user changes the value of the field
     */
    onInput: PropTypes.func,
    /**
     * Example of what the user should type into the field
     */
    placeholder: PropTypes.string,
    /**
     * Marks input as read-only
     */
    readOnly: PropTypes.bool,
    /**
     * Corresponds to the type attribute of an <input>. Relevant for designating a password field, for example.
     */
    type: PropTypes.string,
    /**
     * Value of the field
     */
    value: PropTypes.string
  };

  focus = () => {
    this.input.focus();
  };

  render() {
    return (
      <input
        ref={input => {
          this.input = input;
        }}
        className={cx("hig__text-field-v1__input", {
          "hig__text-field-v1__input--no-value": !this.props.value
        })}
        {...this.props}
      />
    );
  }
}
