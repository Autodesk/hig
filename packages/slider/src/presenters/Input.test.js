import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Input from "./Input";

const subject = (props = {}) => mount(<Input {...props} />);

describe("Slider/presenters/Input", () => {
  let wrapper;
  it("renders correctly without a value", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("event handlers", () => {
    let eventHandler;

    describe("onBlur", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onBlur: eventHandler });
      });

      it("is triggered on blur", () => {
        wrapper.simulate("blur");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onChange", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onChange: eventHandler });
      });

      it("is triggered on change", () => {
        wrapper.simulate("change");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onFocus", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onFocus: eventHandler });
      });

      it("is triggered on focus", () => {
        wrapper.simulate("focus");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onInput", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onInput: eventHandler });
      });

      it("is triggered on input", () => {
        wrapper.simulate("input");
        expect(eventHandler).toBeCalled();
      });
    });
  });
});
