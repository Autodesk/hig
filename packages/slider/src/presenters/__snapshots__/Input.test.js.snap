// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider/presenters/Input renders correctly without a value 1`] = `
<input
  aria-valuemax={undefined}
  aria-valuemin={undefined}
  aria-valuenow={undefined}
  className="hig__slider__input"
  type="range"
/>
`;
